server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java

```diff
@@ -1795,4 +1798,17 @@ public void testReservesBytesDuringPeerRecoveryPhaseOne() throws Exception {
             .mapToLong(n -> n.getIndices().getStore().getReservedSize().getBytes()).sum(), equalTo(0L));
     }
 
+    private void assertGlobalCheckpointIsStableAndSyncedInAllNodes(String indexName, int shardId) throws Exception {
+        long maxSeqNo = NO_OPS_PERFORMED;
+        for (IndicesService indicesService : internalCluster().getDataNodeInstances(IndicesService.class)) {
+            IndexShard shard = indicesService.getShardOrNull(new ShardId(resolveIndex(indexName), shardId));
```

Comment:
This will be `null` if the node doesn't have a copy of the shard. I think it's ok where it's used right now but I'd rather have a null check anyway to avoid trapping someone in the future.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/75898/commits/20585d2aacbe324649fa273d039fc309d5ebc816